lib/utils/template_generator.js

```diff
@@ -27,27 +50,31 @@ class TemplateGenerator {
     }
     let {url, filePath, browse} = ext;
     let tmpFilePath = fs.tmpDir(filePath);
-    console.log(__('Installing template from ' + browse).green);
-    console.log(__('Downloading template...').green);
-
-    fs.mkdirpSync(utils.dirname(tmpFilePath));
-    utils.downloadFile(url, tmpFilePath, (err) => {
-      if (err) {
-        console.error(utils.errorMessage(err).red);
-        console.error('Does the template really exist?'.red);
-        console.error(`Embark's supported templates: https://embark.status.im/templates/`.green);
-        process.exit(1);
-      }
-      utils.extractZip(tmpFilePath, fspath, {
-        map: file => {
-          let fixed_path = file.path.split('/');
-          fixed_path.shift(); // remove first directory
-          file.path = utils.joinPath(...fixed_path);
-          return file;
+    try {
+      try {
+        await this.download(url, tmpFilePath, browse);
+      } catch (err) {
+        let {url_fallback, filePath_fallback, browse_fallback} = ext;
+        if (url_fallback) {
+          console.log(__('Retrying with the master branch...').yellow);
+          tmpFilePath = fs.tmpDir(filePath_fallback);
+          await this.download(url_fallback, tmpFilePath, browse_fallback);
+        } else {
+          throw new Error();
```

Comment:
You should put an error message here, like: "No fallback. Check our templates here: TEMPLATE_URL"

Reply:
It's caught in the outer try/catch and the messages in `downloadFailed` are displayed.

Reply:
Oh right, that's good then.